cron.py

```diff
@@ -0,0 +1,182 @@
+    cl = (s.end - s.start).total_seconds()
+    if ln + cl > lim:
+      if txt:
+        docs.append(create_doc(txt, id, start, end))
+      ln = cl
+      txt = s.content
+      start = s.start
+      end = s.end
+    else:
+      ln += cl
+      txt += " " + s.content
+      end = s.end
+  if txt:
+    docs.append(create_doc(txt, id, start, end))
+  return docs
+
+
+def load_chunks(inventory, lg, ck):
+  chks = []
+  for i, r in inventory.iterrows():
+    try:
+      sr = load_srt(r.id, lg)
+    except HTTPError as _:
+      continue
+    chks += chunk_srt(sr, r.id, lim=ck)
+    if DEV_ENV: break;
+  return chks
+
+
+def load_vector(d):
+  embed = OpenAIEmbeddings()
+  result = embed.embed_query(d.page_content)
+  return result
+
+def select_docs(dt, ch, lg, lm, ck, ct):
+  print("loading chunks...")
+  docs = load_chunks(inventory, lg, ck)
+  docs_list = [(d,) for d in docs]
+
+  print("loading vectors...")
+  with ThreadPool(THREAD_COUNT) as pool:
+    vectors = pool.starmap(load_vector, docs_list)
+
+  print("number of vectors =", len(vectors))
+  kmeans = KMeans(n_clusters=ct, random_state=10, n_init=10).fit(vectors)
+  cent = sorted([np.argmin(np.linalg.norm(vectors - c, axis=1)) for c in kmeans.cluster_centers_])
+  return [docs[i] for i in cent]
+
+
+def id_to_time(id, start=0):
+  dt = IDDTRE.match(id).groups()[0]
+  return datetime.strptime(dt, "%Y%m%d_%H%M%S") + timedelta(seconds=start)
+
+
+def get_summary(d):
+  msg = f"""
+  ```{d}```
+
+  Create the most prominent headline from the text enclosed in three backticks (```) above, describe it in a paragraph, assign a category to it, determine whether it is of international interest, determine whether it is an advertisement, and assign the top three keywords in the following JSON format:
+
+  {{
+    "title": "<TITLE>",
+    "description": "<DESCRIPTION>",
+    "category": "<CATEGORY>",
+    "international_interest": true|false,
+    "advertisement": true|false,
+    "keywords": ["<KEYWORD1>", "<KEYWORD2>", "<KEYWORD3>"]
+  }}
+  """
+  res = openai.ChatCompletion.create(
+    model=MODEL,
+    messages=[{"role": "user", "content": msg}]
+  )
+  result = json.loads(res.choices[0].message.content.strip())
+  result = result | d.metadata
+  result["transcript"] = d.page_content.strip()
+  return result
+
+
+if LM == "Vicuna":
+  openai.api_key = "EMPTY"
+  openai.api_base = VICUNA
+
+# attempt to create ./OUTPUT_FOLDER_NAME
+try:
+  os.mkdir(f"./{OUTPUT_FOLDER_NAME}")
+except FileExistsError:
+  pass
+
+for ch in CHANNELS:
+  print(f"---\nstarting {ch}...")
+  try:
+    print("loading inventory...")
+    inventory = load_inventory(ch, DT, LG)
+  except HTTPError as _:
+    print(f"Inventory for `{ch}` channel is not available for `{DT[:4]}-{DT[4:6]}-{DT[6:8]}` yet, try selecting another date!", icon="⚠️")
+
+  print("loading documents...")
+  seldocs = select_docs(DT, ch, LG, LM, CK, CT)
+
+  print("begin summarizing each document...")
+  if DEV_ENV: seldocs = seldocs[:1]
```

Comment:
I think this can go away (and corresponding initialization too).

Reply:
Okay, I removed all instances of `DEV_ENV`